Write transfer analysis

Nothari117 opened the pull request:
these are the case analysis statements that i used to write the code.

Nothari117 committed:
Create apb_slave

Nothari117 committed:
Create tb_apb_slave

Nothari117 committed:
Create case_analysis_write_transfer

these are the case that i used for writing the code.do take a look and let me know.

nkanderson commented on src/case_analysis_write_transfer:
```diff
@@ -0,0 +1,37 @@
+
+#### **5. Timing Considerations**
+- The number of wait states should **not exceed the number of cycles PREADY is deasserted**.
+- The master must **not change address or control signals** during the access phase.
```
Is this something we should test, ensuring that `plsverr` is asserted if this occurs?

nkanderson reviewed (changes requested):
Hey @Nothari117, this is looking good! I think the following 3 changes would be helpful:
- Remove the `apb_slave` and `tb_apb_slave` files, since we already have equivalent files from previous PRs
- Move your case analysis into the existing `docs/TESTING.md` file
- Update the case analysis text to use the terms we've been using:
  - Change `slave` to `peripheral`
  - Change `master` to `bridge` or `requester`

Nothari117 committed:
case analysis edited

added the case analysis to the testing.md file

Co-Authored-By: nkanderson <4128825+nkanderson@users.noreply.github.com>
Co-Authored-By: GowthamiMedarametla <187230978+GowthamiMedarametla@users.noreply.github.com>
Co-Authored-By: tnl3pdx <98623531+tnl3pdx@users.noreply.github.com>

Nothari117 committed:
added tasks for write

Co-Authored-By: nkanderson <4128825+nkanderson@users.noreply.github.com>
Co-Authored-By: GowthamiMedarametla <187230978+GowthamiMedarametla@users.noreply.github.com>
Co-Authored-By: tnl3pdx <98623531+tnl3pdx@users.noreply.github.com>

Nothari117 committed:
tasks_for_write

made the requested changes, do let me if further changes are needed

Co-Authored-By: nkanderson <4128825+nkanderson@users.noreply.github.com>
Co-Authored-By: GowthamiMedarametla <187230978+GowthamiMedarametla@users.noreply.github.com>
Co-Authored-By: tnl3pdx <98623531+tnl3pdx@users.noreply.github.com>

Nothari117 committed:
task test_write edits

Co-Authored-By: GowthamiMedarametla <187230978+GowthamiMedarametla@users.noreply.github.com>
Co-Authored-By: nkanderson <4128825+nkanderson@users.noreply.github.com>
Co-Authored-By: tnl3pdx <98623531+tnl3pdx@users.noreply.github.com>

Nothari117 committed:
assertions and codes for future additions

we do not want these new files to interact with the existing working code but want these files to exist in the main branch.

Co-Authored-By: GowthamiMedarametla <187230978+GowthamiMedarametla@users.noreply.github.com>
Co-Authored-By: tnl3pdx <98623531+tnl3pdx@users.noreply.github.com>
Co-Authored-By: nkanderson <4128825+nkanderson@users.noreply.github.com>
